perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -1476,20 +1465,17 @@ public Application rejectApplication(PerunSession sess, int appId, String reason
 			log.info("Application {} marked as REJECTED.", appId);
 
 			// get all reserved logins
-			List<Pair<String, String>> logins = jdbc.query("select namespace,login from application_reserved_logins where app_id=?",
-					(resultSet, arg1) -> new Pair<>(resultSet.getString("namespace"), resultSet.getString("login")), appId);
-
-			// delete passwords for reserved logins
+			List<Pair<String, String>> logins = usersManager.getReservedLoginsOnlyByGivenApp(sess, app.getId());
```

Comment:
I think we are repeating this part of code in deleteApplication() and rejectApplication() and it could be extracted to own private method.